### DISCUSSION TITLE
Should extract.py truncate sequences longer than 1024?

### DISCUSSION THREAD

**Discussion by sbonner0** [facebookresearch/esm#82] [Q&A] [answered]
I have been trying to use` extract.py` to produce embeddings for a series of proteins, however some of them have a greater sequence length than 1024. I understand that during training, longer sequences were simply truncated and I was wondering why this does not happen in `extract.py`?

**Comment by tomsercu** [comment, marked as answer]
See the discussion in #21 . There's also some discussion of the different positional embedding types in #49 .
The convenience script `extract.py` does not support those options, but feel free to send a PR!

**Reply by sbonner0** [reply]
Thanks for the reply! I'll look into the links - but am I right then in assuming that the sequences should be truncated before being passed to extract.py? Else any sequence over that will cause the script to crash.

**Reply by tomsercu** [reply]
Yes that's right. I realize now that may be not a great user experience 😕. But silently doing something unexpected like truncating is arguably worse.

**Reply by sbonner0** [reply]
Thanks - I have made a PR with the ability to optionally truncate the sequences.
